### ISSUE TITLE
`UNSUPPORTED_PLATFORMS` and `XFAIL_PLATFORMS` leak `CMAKE_SYSTEM_PROCESSOR` values

### ISSUE THREAD

**Issue by bjacob** [iree-org/iree#12692]
Some of our CMake test-generating functions have an optional `UNSUPPORTED_PLATFORMS` field allowing to specify target architectures where to skip the test. The values recognized there are those produced by `iree_get_platform`, which returns `CMAKE_SYSTEM_PROCESSOR` in the generic case. This means that if `UNSUPPORTED_PLATFORMS` is used in practice, it will result in hardcoding `CMAKE_SYSTEM_PROCESSOR` values in our build/test setup.  But `CMAKE_SYSTEM_PROCESSOR` are nonsensical platform-specific oddities, and some platforms don't even define that at all.

Fortunately, `UNSUPPORTED_PLATFORMS` is currently unused, according to `git grep UNSUPPORTED_PLATFORMS`. So we can easily take either route:
1. Either drop `UNSUPPORTED_PLATFORMS`.
2. Or let `iree_get_platform` use `IREE_ARCH` introduced in #12687.

**Comment by bjacob** [comment]
Ah, I was missing `XFAIL_PLATFORMS`. That one is used in build_tools/python/cmake_builder/rules_test.py with values
`arm` and `ppc` so that would need to be updated.

**Comment by bjacob** [comment]
Ah but `rules_test` is just a test, it doesn't count as usage.

I say we YAGNI this one away. Big simplification.

**Comment by bjacob** [comment]
Or if we want to keep this, just reimplement `iree_get_platform` as:

```
function(iree_get_platform PLATFORM)
  set(${PLATFORM} "${CMAKE_SYSTEM_NAME}-${IREE_ARCH}" PARENT_SCOPE)
endfunction()
```

This way we consistently get the OS name and the architecture, and we have IREE_ARCH consistent stable names for the architecture.

**Comment by GMNGeoffrey** [comment]
I think it's one of those things that isn't used much in the steady state, but is used when something is introduced failing on some platforms or temporarily disabled on them.

I think there is real usage though: https://cs.github.com/openxla/iree/blob/663f92cd71ed3a85ca677e847d2446ff48423080/build_tools/python/e2e_model_tests/test_definitions.py?q=UNSUPPORTED_PLATFORMS#L69

**Comment by bjacob** [comment]
Tentative fix in #12699
